helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -67,14 +68,30 @@
     // @formatter:on
   }
 
-  private final HelixZkClient zkClient;
+  private final HelixZkClient _zkClient;
+  // true if ConfigAccessor was instantiated with a HelixZkClient, false otherwise
+  // This is used for close() to determine how ConfigAccessor should close the underlying ZkClient
+  private boolean _usesExternalZkClient;
 
   /**
    * Initialize an accessor with a Zookeeper client
+   * Note: it is recommended to use the other constructor instead to avoid having to create a HelixZkClient.
    * @param zkClient
    */
   public ConfigAccessor(HelixZkClient zkClient) {
-    this.zkClient = zkClient;
+    this._zkClient = zkClient;
```

Comment:
This "this" is not necessary after we change the name to _zkClient.

Reply:
Removed.